packages/eslint-config-typescript/react.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * strvcom/eslint-config-typescript
+ *
+ * @author      Danny Kijkov <dankijkov@gmail.com>
+ * @copyright   2021 STRV
+ * @license     http://choosealicense.com/licenses/bsd-3-clause  BSD-3-Clause License
+ */
+
+'use strict'
+
+module.exports = {
+
+  rules: {
+    'react/prop-types': 'off',
+
+    '@typescript-eslint/consistent-type-definitions': 'off',
+
+    '@typescript-eslint/naming-convention': [
+      'warn',
+      {
+        selector: 'memberLike',
+        modifiers: ['private'],
+        format: ['camelCase'],
+        leadingUnderscore: 'forbid',
+      },
+      {
+        selector: 'variable',
+        types: ['boolean'],
+        format: ['PascalCase'],
+        prefix: ['is', 'should', 'has', 'can', 'did', 'will', 'expected'],
+      },
+      {
+        selector: 'typeParameter',
+        format: ['PascalCase'],
+        custom: {
+          regex: '^T[A-Z][a-zA-Z]+',
+          match: true,
+        },
+      },
+      {
+        selector: 'interface',
+        format: ['PascalCase'],
+        custom: {
+          regex: '^I[A-Z]',
+          match: true,
+        },
+      },
+    ],
+
+    '@typescript-eslint/no-shadow': 'off',
```

Comment:
You really want to turn this off? This has the potential to lead to a lot of scoping errors where a variable with the same name is declared in a different scope but your intention is to reference the value of that upper scope, see [MDN for details](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const#block_scoping).

This feels like too great a risk just so you can use `name`. 🤷‍♂️ Perhaps some of the rule's options could be tuned to allow that use case but still keep the rule checking for other programmer mistakes?

Reply:
I generally like the rule and agree it's a kinda a great risk.

Tho on frontend there are just too many common names (`event`, `Document`, `location`, `origin` and others). Of course, we can [allow](https://eslint.org/docs/rules/no-shadow#allow) exceptions for those names. But I am afraid we won't create a complete list, so sooner or later the rule would be turned off completely. 

It's simply a very annoying rule a ppl prefer to turn it off rather than leverage its benefits. 

I believe there are other ways to check those possible mistakes, rather than forcing some naming.

Reply:
I believe the alternative would be to disable the `browser: true` environment config so that ESLint no longer thinks that this code is run in a browser, therefore there would be no `event`, `Document`, `location` or any other browser-related globals available.

On one hand, `no-shadow` would no longer complain, on the other hand, accessing those globals would be treated as accessing an undefined variable. Not sure if this is an acceptable trade-off. 🤔

Reply:
I think disabling `browser: true` will lead to more issues than we want 😅

Reply:
So how do you differentiate between a global `Document` object or a let's say `Document` component?